src/os/bluestore/BlueStore.cc

```diff
@@ -8605,6 +8605,19 @@ int BlueStore::dump_bluefs_sizes(ostream& out)
   return r;
 }
 
+void BlueStore::trim_free_space(const string& type)
+{
+  auto iterated_allocation = [&](size_t off, size_t len) {
+    ceph_assert(len > 0);
+    bdev->discard(off, len);
+  };
+  if (type == "bdev-block" && bdev->is_discard_supported()) {
```

Comment:
I think check `bdev->is_discard_supported()` warrants separate message like:
"device 'bdev-block' does not support discard" to cout.